### DISCUSSION TITLE
Interactive financial charts?

### DISCUSSION THREAD

**Discussion by gryffus** [GSI-CS-CO/chart-fx#658] [General]
Hello, i would like to accomplish following:

I have a JavaFX GUI application for analyzing market data
I would like to show a candlestick chart with data I have
The user should have possibility to draw fibonacci retracement on the chart
I would then like to get the coordinates of the fibonacci

But I have not found any mentions about interactivity of ChartFX charts, so my question is, obviously:

Does the ChartFX support user input or drawing in the chart canvas, preferably even with some "snap-to-chart" feature?
Are there any examples of this (preferably with the financial charts implementation)?

Thank you very much for any hints!

**Comment by wirew0rm** [comment]
Hey,

while this is certainly not something chart-fx can do out of the box, you can have a look at the various Plugins which implement similar things and then build your own plugin according to your needs. I'd point you towards the [EditDataSet](https://github.com/fair-acc/chart-fx/blob/main/chartfx-chart/src/main/java/io/fair_acc/chartfx/plugins/EditDataSet.java) and [DataPointTooltip](https://github.com/fair-acc/chart-fx/blob/main/chartfx-chart/src/main/java/io/fair_acc/chartfx/plugins/DataPointTooltip.java) plugins, which should show how you how to register to mouse events on the chart, get corresponding dataset points and draw data on top or below of the charts.

Feel free to ask for help here if you encounter problems with that or share what you built so others can learn and build upon.

**Comment by gryffus** [comment]
Amazing, thanks for the reference.

Do you think it would be feasible to implement "Drawing" directly in the financial charts or do just really a separate plugin? Or somehow even split it to enable drawing for different chart types too? How would you basically recommend to implement it so it could be accepted as a SR?

Thanks again!

**Reply by RalphSteinhagen** [reply]
@gryffus thanks for your interest in ChartFx and perhaps providing a PR for this. We had some early prototypes drawing splines, straight lines, etc., and there are a lot of use cases. Unfortunately, we did/do not have the time to follow up on a full production-grade implementation, but we are happy to advise and assist.

There are a few corner cases regarding drawing/modifying graphs, but the existing infrastructure should allow for it.  IMO the best approach would be to write this as a Plugin. This would allow users to easily opt in/out and provide the most flexibility.

Let us know if you are interested in contributing and we could setup a video-call or coordinate via our matrix channel here: https://matrix.to/#/#fair-acc_chart:gitter.im
